blockchain/setup.go

```diff
@@ -1,210 +1,121 @@
+		return fmt.Errorf("failed to get session for %s, %s: %s", setup.OrgName, setup.OrgAdmin, err)
 	}
-	setup.Client = client
+	orgAdminUser := session
 
-	// Make a new instance of channel pre-configured with the info we have provided,
-	// but for now we can't use this channel because we need to create and
-	// make some peer join it
-	channel, err := fcutil.GetChannel(setup.Client, setup.ChannelId)
-	if err != nil {
-		return nil, fmt.Errorf("Create channel (%s) failed: %v", setup.ChannelId, err)
-	}
-	setup.Channel = channel
-
-	// Get an orderer user that will validate a proposed order
-	// The authentication will be made with local certificates
-	ordererUser, err := fcutil.GetPreEnrolledUser(
-		client,
-		"ordererOrganizations/example.com/users/Admin@example.com/keystore",
-		"ordererOrganizations/example.com/users/Admin@example.com/signcerts",
-		"ordererAdmin",
-	)
-	if err != nil {
-		return nil, fmt.Errorf("Unable to get the orderer user failed: %v", err)
-	}
-
-	// Get an organisation user (admin) that will be used to sign the proposal
-	// The authentication will be made with local certificates
-	orgUser, err := fcutil.GetPreEnrolledUser(
-		client,
-		"peerOrganizations/org1.example.com/users/Admin@org1.example.com/keystore",
-		"peerOrganizations/org1.example.com/users/Admin@org1.example.com/signcerts",
-		"peerorg1Admin",
-	)
-	if err != nil {
-		return nil, fmt.Errorf("Unable to get the organisation user failed: %v", err)
+	// Create channel
+	req := chmgmt.SaveChannelRequest{ChannelID: setup.ChannelID, ChannelConfig: setup.ChannelConfig + "chainhero.channel.tx", SigningIdentity: orgAdminUser}
+	if err = chMgmtClient.SaveChannel(req); err != nil {
+		return fmt.Errorf("failed to create channel: %v", err)
 	}
 
-	// Initialize the channel "mychannel" based on the genesis block by
-	//   1. locating in fixtures/channel/mychannel.tx and
-	//   2. joining the peer given in the configuration file to this channel
-	if err := fcutil.CreateAndJoinChannel(client, ordererUser, orgUser, channel, setup.ChannelConfig); err != nil {
-		return nil, fmt.Errorf("CreateAndJoinChannel return error: %v", err)
-	}
-
-	// Give the organisation user to the client for next proposal
-	client.SetUserContext(orgUser)
+	// Allow orderer to process channel creation
+	time.Sleep(time.Second * 5)
 
-	// Setup Event Hub
-	// This will allow us to listen for some event from the chaincode
-	// and act on it. We won't use it for now.
-	eventHub, err := getEventHub(client)
+	// Org resource management client
+	setup.admin, err = setup.sdk.NewClient(fabsdk.WithUser(setup.OrgAdmin)).ResourceMgmt()
 	if err != nil {
-		return nil, err
-	}
-	if err := eventHub.Connect(); err != nil {
-		return nil, fmt.Errorf("Failed eventHub.Connect() [%s]", err)
+		return fmt.Errorf("failed to create new resource management client: %v", err)
 	}
-	setup.EventHub = eventHub
 
-	// Tell that the initialization is done
-	setup.Initialized = true
+	// Org peers join channel
```

Comment:
Explain in more detail. What happen here?

Reply:
@MaximeAubanel

Reply:
Done